citools/dump_failures.py

```diff
@@ -20,14 +20,22 @@
 RESULTS: Dict[str, Dict[str, Dict[str, List[int]]]] = defaultdict(
     lambda: defaultdict(lambda: defaultdict(lambda: [0, 0]))
 )
+
 SKIPPED_MANIFESTS = []
 
 
-def update_skipped_manifest():
-    path_list = pathlib.Path("dom/canvas/test").glob("**/*.ini")
-    for skip_manifest in path_list:
-        if skip_manifest.name == "mochitest.ini":
-            SKIPPED_MANIFESTS.append(skip_manifest)
+def find_related_manifest(path):
+    try:
+        # make absolute path in case this has symlinks
+        path_absolute = pathlib.Path(path).resolve(strict=True).parent
+        SKIPPED_MANIFESTS.append(
+            found_manifest
+            for found_manifest in pathlib.Path(path_absolute).glob("**/*.ini")
+        )
+    except FileNotFoundError:
+        SKIPPED_MANIFESTS.append(path)
+    except RuntimeError:
+        SKIPPED_MANIFESTS.append(path)
```

Comment:
where do you call find_related_manifest("dom/canvas/test") ?  I want to make sure we don't lose our reference to a known manifest we need to skip

Reply:
I'm implementing the units tests, where I call the function find_related_manifest(path), with the path of the example "dom/canvas/test". I'm trying to mock the "pathlib.path.resolve()" function to generate the exception FileNotFoundError.

Reply:
it sounds like you are handling this correctly, I appreciate the good coding practices.  We do need in dump failures to make sure that SKIPPED_MANIFESTS has `dom/canvas/test/mochitest.ini`